tests/test_freqtradebot.py

```diff
@@ -2526,13 +2530,101 @@ def test_may_execute_sell_after_stoploss_on_exchange_hit(default_conf,
     })
     mocker.patch('freqtrade.exchange.Exchange.get_order', stoploss_limit_executed)
 
-    freqtrade.process_maybe_execute_sell(trade)
+    freqtrade.process_maybe_execute_sells(trades)
     assert trade.stoploss_order_id is None
     assert trade.is_open is False
     assert trade.sell_reason == SellType.STOPLOSS_ON_EXCHANGE.value
     assert rpc_mock.call_count == 2
 
 
+def test_may_execute_sell_stoploss_on_exchange_multi(default_conf,
+                                                     ticker, fee,
+                                                     limit_buy_order,
+                                                     markets, mocker) -> None:
+    default_conf['max_open_trades'] = 3
+    default_conf['exchange']['name'] = 'binance'
+    patch_RPCManager(mocker)
+    patch_exchange(mocker)
+
+    stoploss_limit = {
+        'id': 123,
+        'info': {
+            'foo': 'bar'
+        }
+    }
+    stoploss_order_open = {
+        "id": "123",
+        "timestamp": 1542707426845,
+        "datetime": "2018-11-20T09:50:26.845Z",
+        "lastTradeTimestamp": None,
+        "symbol": "BTC/USDT",
+        "type": "stop_loss_limit",
+        "side": "sell",
+        "price": 1.08801,
+        "amount": 90.99181074,
+        "cost": 0.0,
+        "average": 0.0,
+        "filled": 0.0,
+        "remaining": 0.0,
+        "status": "open",
+        "fee": None,
+        "trades": None
+    }
+    stoploss_order_closed = stoploss_order_open.copy()
+    stoploss_order_closed['status'] = 'closed'
+    # Sell first trade based on stoploss, keep 2nd and 3rd trade open
+    stoploss_order_mock = MagicMock(
+        side_effect=[stoploss_order_closed, stoploss_order_open, stoploss_order_open])
+    create_sl_mock = MagicMock(return_value=True)
+
+    mocker.patch('freqtrade.exchange.Binance.stoploss_limit', stoploss_limit)
+    mocker.patch.multiple(
+        'freqtrade.exchange.Exchange',
+        get_ticker=ticker,
+        get_fee=fee,
+        markets=PropertyMock(return_value=markets),
+        symbol_amount_prec=lambda s, x, y: y,
+        symbol_price_prec=lambda s, x, y: y,
+        get_order=stoploss_order_mock,
+    )
+    # Sell 3rd trade (not called for the first trade)
+    handle_trade_mock = MagicMock(side_effect=[False, True])
```

Comment:
> Ready for review and merge, I guess...

i know and i did review your changes for the most part already so they only need "one last look" ... but i'm not yet content with the test i wrote.

It mocks out `handle_trade()`, and because of this, ony the first trade object is updated correctly, we cannot verify if the other 2 are updated correctly, making the return values of this function irrelevant to the test outome.

I'll try to improve this today or tomorrow, then this should be ready to go.

Reply:
> i know and i did review your changes for the most part already so they only need "one last look" ... but i'm not yet content with the test i wrote.

My 'ready for merge' statement was just a short description of the current state from my side, there can then always be several, more than one, further iterations of changes/reviewing, this is the normal part of the development process, we both know it and should not explain it to each other... 🍻

Reply:
yet - its better to be clear as to why i think it's not ready yet than leaving you (and potential observers) in the dark

Reply:
I would appreciate if you could have a look through this test - if the test (as it is) is ok for you we're good to go